proxy/http/HttpSM.cc

```diff
@@ -665,6 +665,20 @@ HttpSM::state_read_client_request_header(int event, void *data)
     state = PARSE_RESULT_ERROR;
   }
 
+#if TS_HAS_TLS_EARLY_DATA
+  SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(netvc);
+  if (ssl_vc != nullptr && ssl_vc->read_from_early_data) {
+    ssl_vc->read_from_early_data = false;
+    // We check whether the methods are safe according to section 9.1.1
+    // from here https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html
+    if (!HttpTransactHeaders::is_method_safe(t_state.hdr_info.client_request.method_get_wksidx())) {
+      SMDebug("http", "client request was from early data but is NOT safe");
+      call_transact_and_set_next_state(HttpTransact::Forbidden);
```

Comment:
This go with the No.3 technique described in https://tools.ietf.org/html/rfc8470#section-3, right?
It'd be great if we can return "425 (Too Early)" for client to clarify that client can retry the request without early data.

>    When a server enables early data, there are a number of techniques it
   can use to mitigate the risks of replay:
> ...
>    3.  The server can cause a client to retry individual requests and
       not use early data by responding with the 425 (Too Early) status
       code (Section 5.2) in cases where the risk of replay is judged
       too great.

Reply:
Yes, I've been working on this.